oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java

```diff
@@ -229,6 +238,9 @@ private ImpersonatedCredentials(Builder builder) {
     if (this.lifetime > TWELVE_HOURS_IN_SECONDS) {
       throw new IllegalStateException(LIFETIME_EXCEEDED_ERROR);
     }
+    if (this.lifetime == 0) {
+      this.lifetime = DEFAULT_LIFETIME_IN_SECONDS;
```

Comment:
If we're doing this client side, perhaps it would be better to default this in the builder.

Reply:
moved to builder